src/tests/test_batch_processing.py

```diff
@@ -7,50 +7,99 @@
+    def test_process_pids(self, capsys, caplog):
+        caplog.set_level('DEBUG')
         batch_processor = BatchProcessor()
-        pids = ["1", "2", "3"]
+        pids = ["A", "B", "C"]
         callback = lambda pid: print(pid)
         batch_processor.process_pids(pids, callback)
         captured = capsys.readouterr()
-        assert captured.out == "1\n2\n3\n"
+        assert captured.out == "A\nB\nC\n"
+        assert len(caplog.records) == 7
+        assert (caplog.records[0].message == 'Start batch processing on 3 entries')
+        assert (caplog.records[1].message == 'Processing 1 of 3 entries: A')
+        assert (caplog.records[2].message == 'Waiting 0.1 seconds before processing next entry')
+        assert (caplog.records[3].message == 'Processing 2 of 3 entries: B')
+        assert (caplog.records[4].message == 'Waiting 0.1 seconds before processing next entry')
+        assert (caplog.records[5].message == 'Processing 3 of 3 entries: C')
+        assert (caplog.records[6].message == 'Batch processing ended: 3 entries processed')
 
-    def test_process_objects_with_pids(self, capsys):
+    def test_process_dicts_with_pids(self, capsys, caplog):
+        caplog.set_level('DEBUG')
         batch_processor = BatchProcessor()
-        objects = [{"PID": "1", "param": "value1"},
-                   {"PID": "2", "param": "value2"},
-                   {"PID": "1", "param": "value3"}]
+        objects = [{"PID": "a", "param": "value1"},
+                   {"PID": "b", "param": "value2"},
+                   {"PID": "a", "param": "value3"}]
         batch_processor.process_pids(objects, lambda obj: print(obj))
         captured = capsys.readouterr()
-        assert captured.out == ("{'PID': '1', 'param': 'value1'}\n"
-                                "{'PID': '2', 'param': 'value2'}\n"
-                                "{'PID': '1', 'param': 'value3'}\n")
+        assert captured.out == ("{'PID': 'a', 'param': 'value1'}\n"
+                                "{'PID': 'b', 'param': 'value2'}\n"
+                                "{'PID': 'a', 'param': 'value3'}\n")
+        assert len(caplog.records) == 7
+        assert (caplog.records[0].message == 'Start batch processing on 3 entries')
+        assert (caplog.records[1].message == 'Processing 1 of 3 entries: a')
+        assert (caplog.records[2].message == 'Waiting 0.1 seconds before processing next entry')
+        assert (caplog.records[3].message == 'Processing 2 of 3 entries: b')
+        assert (caplog.records[4].message == 'Waiting 0.1 seconds before processing next entry')
+        assert (caplog.records[5].message == 'Processing 3 of 3 entries: a')
+        assert (caplog.records[6].message == 'Batch processing ended: 3 entries processed')
 
-    def test_process_objects_without_pids(self, capsys):
+    def test_nothing_to_process(self, capsys, caplog):
+        caplog.set_level('DEBUG')
+        batch_processor = BatchProcessor()
+        objects = []
```

Comment:
Expected this would cover https://app.codecov.io/gh/DANS-KNAW/dans-datastation-tools/pull/55/blob/src/datastation/common/batch_processing.py#L73